### PR TITLE
cumulus-consensus-common: block import: `delayed_best_block` flag added

### PR DESCRIPTION
**Pull request by michalkucharczyk**
This PR adds the  `delayed_best_block` flag to `ParachainBlockImport`. If not set, the `params.fork_choice` is not updated (to `ForkChoiceStrategy::Custom`) during the block import. 

When `delayed_best_block` is set to `false`  all parachain blocks on the [longest fork](https://github.com/paritytech/polkadot-sdk/blob/552be4800d9e4b480f79a300fc531783e04be099/substrate/client/service/src/client/client.rs#L708-L709) will be notified as the best block, allowing transaction pool to be updated with every imported block.

Otherwise imported blocks will  not be notified as best blocks (`fork_choice=ForkChoiceStrategy::Custom(false)`), and transaction pool would be updated only with best block received from relay-chain.

Improvement for: #1202

### CODE REVIEW

**Comment by bkchr** [comment]
@michalkucharczyk we should get this merged.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4417902

File: cumulus/client/consensus/common/src/lib.rs [hunk 1]
```diff
@@ -111,12 +111,11 @@ impl<B: BlockT> ParachainConsensus<B> for Box<dyn ParachainConsensus<B> + Send +
 
 /// Parachain specific block import.
 ///
-/// This is used to set `block_import_params.fork_choice` to `false` as long as the block origin is
-/// not `NetworkInitialSync`. The best block for parachains is determined by the relay chain.
-/// Meaning we will update the best block, as it is included by the relay-chain.
+/// Specialized block import for parachains. It supports to delay setting the best block until the relay chain has included a candidate in its best block. By default the delayed best block setting is disabled. The block import also monitors the imported blocks and prunes by default if there are too many blocks at the same height. Too many blocks at the same height can for example happen if the relay chain is rejecting the parachain blocks in the validation.
```
Comment (davxy) [inline cumulus/client/consensus/common/src/lib.rs:114]: can you split this doc in multiple line?

**Review by skunert — APPROVED** [review]

**Review by davxy — APPROVED** [review]
LGTM.

But is not clear what is the use case. Can you add a usage scenario to the PR description?

**Pull request merged**